src/lsp/cobol_cfg/cfg_builder.ml

```diff
@@ -0,0 +1,486 @@
+(******************************************************************************)
+(*                                                                            *)
+(*     Copyright (c) 2021-2023 OCamlPro SAS                                   *)
+(*                                                                            *)
+(*     All rights reserved.                                                   *)
+(*     This file is distributed under the terms of the                        *)
+(*     OCAMLPRO-NON-COMMERCIAL license.                                       *)
+(*                                                                            *)
+(******************************************************************************)
+
+open Cobol_unit
+open Cobol_common.Srcloc.INFIX
+open Cobol_common.Srcloc.TYPES
+open Cobol_unit.Types
+open Cobol_common.Visitor
+module NEL = Cobol_common.Basics.NEL
+
+module Options = struct
```

Comment:
I'd rather not depend on `yojson` here if possible. This `type option` should just be exposed by `Cobol_cfg` and constructed by "users" of that library. This means at the moment parsing the JSON should only be done in `Cobol_lsp` I think.